code/weapon/weapons.cpp

```diff
@@ -5078,15 +5078,19 @@ int weapon_create( vec3d * pos, matrix * porient, int weapon_type, int parent_ob
 		size_t *position = get_pointer_to_weapon_fire_pattern_index(weapon_type, parent_shipp, src_turret);
 		Assertion( position != NULL, "'%s' is trying to fire a weapon that is not selected", Ships[parent_objp->instance].ship_name );
 
+		size_t curr_pos = *position;
+		if (((Player_ship->flags[Ship::Ship_Flags::Secondary_dual_fire]) || (Player_ship->flags[Ship::Ship_Flags::Primary_linked])) && (curr_pos > 0)) {
```

Comment:
It is perhaps cleaner to use Player_ship->flags[Ship::Ship_Flags::Secondary_dual_fire, Ship::Ship_Flags::Primary_linked] here. Certainly shorter :)